trollsched/utils.py

```diff
@@ -0,0 +1,150 @@
+    """
+    for k, v in u.items():
+        if isinstance(v, Mapping):
+            r = recursive_dict_update(d.get(k, {}), v)
+            d[k] = r
+        else:
+            d[k] = u[k]
+    return d
+
+def read_config(filename):
+    try:
+        return read_config_yaml(filename)
+    except yaml.parser.ParserError as e:
+        return read_config_cfg(filename)
+
+def read_config_cfg(filename):
+    """Read the config file *filename* and replace the values in global
+    variables.
+    """
+    station_list = []
+    cfg = ConfigParser()
+    cfg.read(filename)
+
+    stations = cfg.get("default", "station").split(",")
+    forward = cfg.getint("default", "forward")
+    start = cfg.getfloat("default", "start")
+
+    pattern = {}
+    for k, v in cfg.items("pattern"):
+        pattern[k] = v
+
+    for station in stations:
+        station_name = cfg.get(station, "name")
+        station_lon = cfg.getfloat(station, "longitude")
+        station_lat = cfg.getfloat(station, "latitude")
+        station_alt = cfg.getfloat(station, "altitude")
+
+        area = resample_utils.parse_area_file(cfg.get(station, "area_file"),
+                                                        cfg.get(station, "area"))[0]
+
+        satellites = cfg.get(station, "satellites").split(",")
+
+        sat_scores = {}
+        for sat in satellites:
+            sat_scores[sat] = (cfg.getfloat(sat, "night"),
+                               cfg.getfloat(sat, "day"))
+
+        station_list.append(((station_lon, station_lat, station_alt),
+                station_name, area, sat_scores))
+
+    return (station_list, forward, start, pattern)
+
+def read_config_yaml(filename):
+    """Read the yaml file *filename* and replace the values in global
+    variables.
+    """
+    station_list = []
+    cfg = read_yaml_file(filename)
+
+    stations = cfg["default"]["station"]
+    forward = cfg["default"]["forward"]
+    start = cfg["default"]["start"]
+
+    pattern = {}
+    for k, v in cfg["pattern"].items():
+        pattern[k] = v
+
+    for station in stations:
+        station_name = cfg["stations"][station]["name"]
+        station_lon = cfg["stations"][station]["longitude"]
+        station_lat = cfg["stations"][station]["latitude"]
+        station_alt = cfg["stations"][station]["altitude"]
+
+        area = resample_utils.parse_area_file(cfg["stations"][station]["area_file"],
+                                              cfg["stations"][station]["area"])[0]
+
+        satellites = cfg["stations"][station]["satellites"]
+
+        sat_scores = {}
+        for sat in satellites:
+            sat_scores[sat] = (cfg["satellites"][sat]["night"],
+                                             cfg["satellites"][sat]["day"])
+
+        station_list.append(((station_lon, station_lat, station_alt),
+                station_name, area, sat_scores))
+
+    foo = cfg["default"].get("center")
+    if foo:
+        schedule.CENTER_ID = foo
```

Comment:
This doesn't seem to be used in the `generate_schedule_xml` function

Reply:
It doesn't indeed. I've used the sort-of static schedule.CENTER_ID to stay compatible with the cfg-file, where the center_id wasn't set.
With your commit in develop these lines might not be necessary?

Reply:
Well, I pass center_id as an argument, I think your approach is superior.